pkg/tiller/resource_policy.go

```diff
@@ -29,11 +33,16 @@ const resourcePolicyAnno = "helm.sh/resource-policy"
 //   during an uninstallRelease action.
 const keepPolicy = "keep"
 
-func filterManifestsToKeep(manifests []Manifest) ([]Manifest, []Manifest) {
+func filterManifestsToKeep(manifests []Manifest, kubeClient environment.KubeClient, namespace string) ([]Manifest, []Manifest) {
 	remaining := []Manifest{}
 	keep := []Manifest{}
 
 	for _, m := range manifests {
+		// check if m is in fact present from k8s client's POV.
+		output, err := kubeClient.Get(namespace, bytes.NewBufferString(m.Content))
```

Comment:
I worry about this taking a while with large charts because it is an individual API call for each resource, even if it is not one we are keeping. This may go better in `summarizeKeptManifests`. What do you think?

Reply:
Sure makes total sense. I've pushed the changes, the only problem now is that the message header is still shown even if all the supposedly kept manifest are not there, so you think it's worth to check for this after returning from `summarizeKeptManifests`?

Reply:
Yeah, it might be worth checking if the code isn't too bad

Reply:
Done, PTAL.